routes/webhook/github.rb

```diff
@@ -79,26 +79,28 @@ def handle_workflow_job(data)
       return error("A workflow_job without runner_id")
     end
 
-    runner = GithubRunner.first(
-      installation_id: installation.id,
-      repository_name: data["repository"]["full_name"],
-      runner_id: runner_id
-    )
+    DB.transaction do
+      runner = GithubRunner.for_update.first(
```

Comment:
It's one of our busiest endpoints. Since the operation in the transaction is very quick, `for_update` doesn't increase its latency, right?

Reply:
I just wanted to make sure.

Reply:
> Is there a reason to create a semaphore for a strand that doesn't exist?

No. Also, it should be prevented by a foreign key.

https://github.com/ubicloud/ubicloud/blob/ac85e59d9bffc0e5932278c2a0edd87b627f2083/model/semaphore.rb#L25

Reply:
I came to the same conclusion as you.  I submitted a separate PR to address this issue: #2971

Reply:
I thought we agreed to use `Dataset#update` instead of `for_update` because of the latency penalty.

Reply:
We did, sorry I missed that when I removed the three other cases of `for_update`.  I pushed another commit that removes it from the webhook route.